site/_filters/unique-id.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @fileoverview This makes sure the ids of page sections are unique
+ *
+ * This follows a couple of rules:
+ *   - we store each section id in an array
+ *   - if the section id is already in the array, counts its occurences and makes the id unique using that number
+ *   - otherwise just returns the id we passed to the filter
+ *
+ */
+
+const pageIds = [];
+
+/**
+ * @param {string} id of the section
+ * @param {string} url of the page
+ * @return {string} unique id for the section
+ */
+
+function uniqueId(id, url) {
+  const idUrl = `${url}#${id}`;
+
+  if (pageIds.includes(idUrl)) {
+    const count = pageIds.filter(item => {
+      return item === idUrl;
+    }).length;
+
+    pageIds.push(idUrl);
+
+    return `${id}-${count}`;
+  }
+
+  pageIds.push(idUrl);
+  return id;
+}
```

Comment:
I'd simplify this quite a bit. The idea makes sense, but I would leave it to the author to choose correct IDs, as they might also be a simple copy-paste errors. No one will probably build landing pages in a hurry anyway where it would make sense to auto-fix it for them. This would be my approach:

```js
/**
 * @param {string} id of the section
 * @this {any} The eleventy context
 * @return {string} unique id for the section
 */

function ensureUniqueHref(id) {
  const key = `${this.ctx.page.url}#${id}`;

  // Check if there is already an URL with this key, and if not, add it
  // to the set and exit early.
  if (!pageIds.has(key)) {
    pageIds.add(key);
    return id;
  }

  throw new Error(
    `There is another element with the ID ${id} on the page ${this.ctx.page.url}. Please make sure all IDs are unique.`
  );
}
```

So just throwing if an ID duplicates should be fine.